src/redislog.rs

```diff
@@ -0,0 +1,214 @@
+extern crate chrono;
+extern crate r2d2;
+extern crate r2d2_redis;
+extern crate redis;
+
+use chrono::{DateTime, SecondsFormat, Utc};
+use r2d2_redis::RedisConnectionManager;
+use serde_json::json;
+use std::fmt;
+use std::process::Command;
+use std::time::Duration;
+
```

Comment:
This can be moved to its own lib later. Of course I will have to expand on the comments.